onnxruntime/test/providers/cpu/math/gemm_test.cc

```diff
@@ -1038,224 +1037,293 @@ TEST(GemmOpTest, SharedPrepackedWeights) {
-        for (int64_t k = 0; k < K; ++k) {
-          sum += a_data[i * K + k] * b_data[k * N + j];
+  // Calculate expected output based on transpose settings
+  std::vector<float> expected_data(M * N, 0.0f);
+  for (int64_t i = 0; i < M; ++i) {
+    for (int64_t j = 0; j < N; ++j) {
+      float sum = 0.0f;
+      for (int64_t k = 0; k < K; ++k) {
+        float a_val, b_val;
+
+        if (transA) {
+          a_val = a_data[k * M + i];  // A^T[i][k] = A[k][i]
+        } else {
+          a_val = a_data[i * K + k];  // A[i][k]
         }
-        expected_data[i * N + j] = sum + get_bias_value(c_data, bias_type, i, j, N);
+
+        if (transB) {
+          b_val = b_data[j * K + k];  // B^T[k][j] = B[j][k]
+        } else {
+          b_val = b_data[k * N + j];  // B[k][j]
+        }
+
+        sum += a_val * b_val;
       }
+      float matmul_result = alpha * sum;
+      float bias_value = beta * get_bias_value(c_data, bias_type, i, j, N);
+      expected_data[i * N + j] = matmul_result + bias_value;
     }
+  }
 
-    test.AddOutput<float>("Y", {M, N}, expected_data);
-    test.ConfigExcludeEps({kQnnExecutionProvider})
-        .Config(run_with_tunable_op)
-        .RunWithConfig();
-  };
+  test.AddOutput<float>("Y", {M, N}, expected_data);
+
+  test.ConfigExcludeEps({kQnnExecutionProvider})
+      .Config(run_with_tunable_op)
+      .RunWithConfig();
+};
 
-  // Test different matrix sizes with all bias types
+TEST(GemmOpTest, GemmOptimizePacked) {
+  // Test different matrix sizes with all bias types and alpha/beta combinations
   std::vector<std::tuple<int64_t, int64_t, int64_t>> test_sizes = {
-      {32, 32, 32}, {64, 64, 64}, {60, 16, 92}, {8, 8, 8}, {128, 128, 128}, {128, 32, 64}, {96, 24, 48}, {48, 48, 120}, {72, 80, 84}, {33, 67, 99}, {1, 1, 1}, {31, 31, 31}};
+      {32, 32, 32}, {64, 64, 64}, {60, 16, 92}, {8, 8, 8}, {128, 128, 128}, {128, 32, 64}, {96, 24, 48}, {48, 48, 120}, {72, 80, 84}, {33, 67, 99}, {1, 1, 1}, {31, 31, 31}, {37, 64, 256}, {256, 64, 1024}, {1, 64, 448}};
 
   std::vector<BiasType> bias_types = {
       BiasType::noBias, BiasType::MBias, BiasType::ScalarBias,
       BiasType::MNBias, BiasType::NBias};
 
-  // Run tests with different combinations of matrix sizes and bias types
+  std::vector<std::pair<float, float>> alpha_beta_values = {
+      {1.0f, 1.0f}, {0.5f, 1.0f}, {1.0f, 2.0f}, {0.5f, 2.0f}, {2.0f, 0.5f}, {0.0f, 1.0f}, {1.0f, 0.0f}, {2.0f, 2.0f}, {0.0f, 0.0f}};
+
+  // Run tests with different combinations of matrix sizes, bias types, and alpha/beta values
   for (const auto& size : test_sizes) {
     for (const auto& bias_type : bias_types) {
-      run_test(std::get<0>(size), std::get<1>(size), std::get<2>(size), bias_type);
+      for (const auto& [alpha, beta] : alpha_beta_values) {
+        run_gemm_optimize_packed_test(std::get<0>(size), std::get<1>(size), std::get<2>(size), bias_type, alpha, beta, false, false);
+      }
     }
   }
 }
 
 TEST(GemmOpTest, GemmOptimizePackedTransA) {
```

Comment:
Maybe further merge `GemmOptimizePackedTransA/GemmOptimizePackedTransB/GemmOptimizePackedTransAB...` into one test `GemmOptimizePacked` to reuse the code if all the data is same except transpose.
```
  for (const auto& size : test_sizes) {
    for (const auto& bias_type : bias_types) {
      for (const auto& [alpha, beta] : alpha_beta_values) {
        for (const auto& [transA, transB] : transpose_values) {